controllers/user.controller.js

```diff
@@ -22,101 +21,93 @@ exports.create = (req, res) => {
-    if(exist.length > 0){
-        console.log(exist)
-        return res.status(422).send({
-            message: 'Email has already been taken.'
-        })
-    }else{
-        User.create(user)
-        .then((data) => {
-          res.send(data);
-        })
-        .catch((err) => {
-          res.status(500).send({
-            message: err.message || "Some error occurred while creating new user.",
-          });
-        });
+  try {
+    const userUniqueEmailCheck = await User.findAll({
+      where: { email: user.email },
+    });
+    if (userUniqueEmailCheck.length > 0) {
+      return res.status(422).send({
+        message: "Email has already been taken.",
+      });
+    } else {
+      const newUser = await User.create(user);
+      return res.send(newUser);
     }
-  })
-
+  } catch (err) {
+    return res.status(500).send({
+      message: err.message || "Some error occurred while creating new user.",
+    });
+  }
 };
 
-exports.findAll = (req, res) => {
-  User.findAll()
-    .then((data) => {
-      res.send(data);
-    })
-    .catch((err) => {
-      res.status(500).send({
-        message:
-          err.message || "Some error occurred while while looking for users.",
-      });
+exports.findAll = async (req, res) => {
+  try {
+    const users = await User.findAll();
+    res.send(users);
+  } catch (err) {
+    res.status(500).send({
+      message: err.message || "Some error occurred while looking for users.",
     });
+  }
 };
 
-exports.findOne = (req, res) => {
-  const id = req.params.id;
-  User.findByPk(id)
-    .then((data) => {
-      if (data) res.send(data);
-      else {
-        res.status(404).send({
-          message: `Cannot find user with id: ${id}`,
-        });
-      }
-    })
-    .catch((err) => {
-      res.status(500).send({
-        message: err.message || "Some error occurred while looking for user.",
+exports.findOne = async (req, res) => {
+  try {
+    const id = req.params.id;
+    const user = await User.findByPk(id);
+    if (user) return res.send(user);
+    else {
+      return res.status(404).send({
+        message: `Cannot find user with id: ${id}`,
       });
+    }
+  } catch (err) {
+    return res.status(500).send({
+      message: err.message || "Some error occurred while looking for user.",
     });
+  }
 };
 
-exports.update = (req, res) => {
-  const id = req.params.id;
-  User.update(req.body, {
-    where: { id: id },
-  })
-    .then((num) => {
-      if (num == 1) {
-        res.send({
-          message: "User was updated successfully.",
-        });
-      } else {
-        res.send({
-          message: `Cannot update User with id=${id}. Maybe User was not found or req.body is empty!`,
-        });
-      }
-    })
-    .catch((err) => {
-      res.status(500).send({
-        message: "Error updating User with id=" + id,
+exports.update = async (req, res) => {
+  try {
+    const id = req.params.id;
+    const response = await User.update(req.body, {
```

Comment:
You cannot pass whole body to update. Extract object to make sure that you're passing only needed properties